util/Keplr.js

```diff
@@ -134,14 +134,19 @@ class Keplr {
     await this.checkIfInited();
     const signerInstance = this.signer;
     const address = this.internalGetWalletAddress();
-    return async function signer(signMessage) {
-      const data = JSON.parse(signMessage);
-      const { signed, signature } = await signerInstance.sign(address, data);
-      return {
-        signed,
-        signature: Buffer.from(signature.signature, 'base64'),
-        publicKey: Buffer.from(signature.pub_key.value, 'base64'),
-      };
+    return {
+      signAmino: async (_, data) => {
+        const { signatures, ...signed } = await signerInstance.sign(address, data);
+        return {
+          signed: signed.signed,
+          signature: {
+            signature: signed.signature.signature,
+            sequence: signed.signed.sequence,
+            account_number: signed.signed.account_number,
+            pub_key: signed.signature.pub_key,
+          },
+        };
+      },
     };
```

Comment:
Would this work? `this.signer` should already be an compatible signer without extra code
```return this.signer```